Library/Homebrew/formula.rb

```diff
@@ -287,6 +287,14 @@ def version
     active_spec.version
   end
 
+  def update_head_version
+    return unless active_spec.is_a? HeadSoftwareSpec
+    head_resource = head.get_resource
+    if head_resource.downloader.is_a? VCSDownloadStrategy
+      head_resource.version.update_commit(head_resource.downloader.last_commit)
```

Comment:
I think for better robustly you should also check whether the local clone exists. Only update commit when it does.